### PR TITLE
LPS-205549 Ensure escapedHREF is not evaluated in a JS context

### PR DESCRIPTION
**Pull request by dsanz**
This is a followup of https://github.com/liferay-frontend/liferay-portal/pull/3923 where:

- We fix the issue in a different way. More specifically,
  - Preserve the `escapedHREF` value as it comes
  - Avoid evaluating it directly in the JS context as part of the inline `onClick` event handler
- Fix is not applied to the [other requested cases](https://github.com/liferay-frontend/liferay-portal/pull/3923#discussion_r1453157957) because there, `escapedHREF` value comes already escaped, including the `'` character.

### CODE REVIEW

File: portal-web/docroot/html/taglib/aui/button/end.jsp [hunk 1]
```diff
@@ -47,7 +47,8 @@
 					onClick="<%= onClick %>"
 				</c:when>
 				<c:when test="<%= Validator.isNotNull(escapedHREF) %>">
-					onClick="Liferay.Util.navigate('<%= escapedHREF %>')"
+					data-href="<%= escapedHREF %>"
+					onClick="Liferay.Util.navigate(this.getAttribute('data-href'))"
```
Comment (markocikos) [inline portal-web/docroot/html/taglib/aui/button/end.jsp:51]: [nit] We can simplify this a bit, to `Liferay.Util.navigate(this.dataset.href)`. 

[even smaller nit] It's a bit odd to see camelCase `onclick` in raw HTML, but it seems browsers have been fine with it for a long time.

**Review by markocikos — APPROVED** [review]
@dsanz LGTM! Couple of non-blocking nits inline.

File: portal-web/docroot/html/taglib/aui/button/end.jsp (hunk 1 above)
Comment (dsanz) [reply portal-web/docroot/html/taglib/aui/button/end.jsp]: I've force pushed the first proposal. Second one would require a deeper review of `onClick` occurrences

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:forward

**Comment by dsanz** [comment]
ci:report:2536178259

**Comment by dsanz** [comment]
no related failures as per https://github.com/liferay-frontend/liferay-portal/pull/3977#issuecomment-1933593999

**Pull request closed without merging**